### PR TITLE
Show next rotation time in credential UI

### PR DESCRIPTION
**Pull request by ryan-lane**
Add info in the credential UI for the next required rotation date for a credential.

### CODE REVIEW

**Comment by ryan-lane** [comment]
Screen Shot 2020-04-02 at 8 10 04 PM

File: confidant/schema/credentials.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ class CredentialResponse(object):
     permissions = attr.ib(default=dict)
     tags = attr.ib(default=list)
     last_rotation_date = attr.ib(default=None)
```
Comment (irhkang) [inline confidant/schema/credentials.py:23]: not related to this PR but we could remove last_rotation_date if we aren't using it (which I don't think we should given that next_rotation_date has some complex logic)

If you remove it remove the line in the test too

File: confidant/public/modules/resources/views/credential-details.html [hunk 2]
```diff
@@ -146,6 +146,10 @@
         <button type="button" ng-disabled="editableForm.$waiting" ng-click="addTag()" class="btn pull-right">Add Tag</button>
       </span>
     </div>
+    <div class="form-group" ng-show="credential.id">
+      <label for="credentialNextRotationDate">Credential Next Rotation Needed Date</label>
+      <span id="credentialNextRotationDate">{{ credential.next_rotation_date }} ({{ credential.daysTillRotation }} days)</span> <span id="credentialNextRotationDateTooltip" class="glyphicon glyphicon-info-sign" data-toggle="tooltip" title="This date is determined by the last rotation date and the last decryption date."></span>
```
Comment (irhkang) [inline confidant/public/modules/resources/views/credential-details.html:151]: we ok with showing a negative number here for expired creds?

File: confidant/schema/credentials.py (hunk 1 above)
Comment (ryan-lane) [reply confidant/schema/credentials.py:23]: I'm ok with delivering it, even if it's not shown in the UI. We may decide to show it later, or folks may want to use it in other tooling

File: confidant/public/modules/resources/views/credential-details.html (hunk 2 above)
Comment (ryan-lane) [reply confidant/public/modules/resources/views/credential-details.html:151]: yep. I think it's good context that it should have been rotated a while ago.

**Pull request merged**